packages/ai/core/generate-image/generate-image.ts

```diff
@@ -159,29 +161,49 @@ Only applicable for HTTP-based providers.
       ),
     );
     warnings.push(...result.warnings);
+
+    if (result.providerMetadata) {
+      for (const [providerName, metadata] of Object.entries<{
+        images: unknown;
+      }>(result.providerMetadata)) {
+        providerMetadata[providerName] ??= { images: [] };
+        providerMetadata[providerName].images.push(
+          ...result.providerMetadata[providerName].images,
+        );
+      }
+    }
```

Comment:
what about provider metadata that's not `images`?

Reply:
It would be merged into the `result.providerMetadata[providerName]` object, but I haven't seen any meta data yet. I can prepare the code for it already though if you like?